charts/liquibase-migration/templates/manifests.yaml

```diff
@@ -0,0 +1,210 @@
+  template:
+    metadata:
+      name: {{ $config.name }}-liquibase-migration
+      {{- if $config.k8sLabelRef }}
+      labels: {{- include $config.k8sLabelRef $ | nindent 8 }}
+      {{- end }}
+    spec:
+      # Allow migration-liquibase to kill migration-sqlproxy's process
+      shareProcessNamespace: true
+      restartPolicy: Never
+      serviceAccountName: {{ $config.k8sServiceAccount }}
+      # Most apps reference "127.0.0.1" in DB URL but some expect "sqlproxy"
+      hostAliases:
+      - ip: 127.0.0.1
+        hostnames:
+        - sqlproxy
+      {{- if or (($config.proxyCredentialFileMount).secretName) (($config.migrationConfigFileMount).secretName) }}
+      volumes:
+      {{- if (($config.proxyCredentialFileMount).secretName) }}
+      - name: proxy-credentials-volume
+        secret:
+          secretName: {{ $config.proxyCredentialFileMount.secretName }}
+      {{- end }}
+      {{- if (($config.migrationConfigFileMount).secretName) }}
+      - name: migration-configs-volume
+        secret:
+          secretName: {{ $config.migrationConfigFileMount.secretName }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: {{ $config.name }}-migration-sqlproxy
+        image: {{ $config.proxyImage }}:{{ $config.proxyImageTag }}
+        {{- if (($config.proxyCredentialFileMount).secretName) }}
+        volumeMounts:
+        - mountPath: {{ $config.proxyCredentialFileMount.mountFilePath | required "proxyCredentialFileMount.mountFilePath required if proxyCredentialFileMount.secretName provided" }}
+          subPath: {{ $config.proxyCredentialFileMount.mountFilePath | splitList "/" | last }}
+          name: proxy-credentials-volume
+          readOnly: true
+        {{- end }}
+        {{- if $config.proxyContainerConfig }}
+        {{- $config.proxyContainerConfig | toYaml | nindent 8 }}
+        {{- end }}
+        command: {{ $config.proxyShell | toYaml | nindent 10 }}
+        args:
+        - |-
+          /cloud_sql_proxy \
+            -verbose \
+            -max_connections={{ $config.proxyArgsMaxConnections }} \
+            -instances={{ $config.proxyArgsInstances }} \
+            {{- if (($config.proxyCredentialFileMount).secretName) }}
+            -credential_file={{ $config.proxyCredentialFileMount.mountFilePath }} \
+            {{- end }}
+      - name: {{ $config.name }}-migration-liquibase
+        image: {{ $config.migrationImage | required "migrationImage required" }}:{{ $config.migrationImageTag }}
+        {{- if (($config.migrationConfigFileMount).secretName) }}
+        volumeMounts:
+        - mountPath: {{ $config.migrationConfigFileMount.mountFilePath | required "migrationConfigFileMount.mountFilePath required if migrationConfigFileMount.secretName provided" }}
+          subPath: {{ $config.migrationConfigFileMount.mountFilePath | splitList "/" | last }}
+          name: migration-configs-volume
+          readOnly: true
+        {{- end }}
+        env:
+        - name: DB_USERNAME
```

Comment:
nit: Any way to simplify some of this conditional logic, since it looks like it's repeated once for each environment variable?

Maybe populate a variable, $credentialsSecret, with the necessary attributes and use that for each variable?